### ISSUE TITLE
Attachment - Newly uploaded MP4 file removing attachment tag after refresh the page

### ISSUE THREAD

**Issue by kbecciv** [Expensify/App#8331]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
1. Go to https://staging.new.expensify.com and log in with any account
2. Click to add attachment 
3. Select mp4 video and tap Send
4. Refresh the page

## Expected Result:
Newly uploaded MP4 file should show mp4 tag

## Actual Result:
Newly uploaded MP4 file removing attachment tag after refresh the page

## Workaround:
Unknown

## Platform:

Where is this issue occurring?

- Web
- iOS
- Android
- Desktop App
- Mobile Web

**Version Number:** 1.1.46.3

**Reproducible in staging?:** Yes

**Reproducible in production?:** Yes

**Email or phone of affected tester (no customers):** rpriyo@gmail.com

**Logs:** https://stackoverflow.com/c/expensify/questions/4856

**Notes/Photos/Videos:** Any additional supporting documentation

https://user-images.githubusercontent.com/93399543/160263780-3e177791-46ce-496e-ac1d-ce297612a2ba.mp4

**Expensify/Expensify Issue URL:** 

**Issue reported by:** Applause

**Slack conversation:** 

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by mountiny** [comment]
Should be `External`. I have tried it locally and the I can see the preview file name shows for zip/mp4 what I have tried but not for images, which I guess is intentional.

**Comment by dennismunene** [comment]
**PROPOSAL**
All attachments are formatted as either  ``img`` or ``a`` tags , 
We need to updated the regex here  to consider both 

Update 
https://github.com/Expensify/App/blob/9b825b44da3cb3c7605691abced3db2c8ada030d/src/libs/actions/Report.js#L186

To this 

```
   const isLastMessageAttachment = /<\/?(a|img|...)\b[^<>]*>/gi.test(lastActionMessage);
```

output :




https://user-images.githubusercontent.com/4717301/161050129-bac27942-0d48-4c85-b803-b36326d14bd7.mov

**Comment by parasharrajat** [comment]
what if we send a link in the message? Will it work? try with `https://google.com`

**Comment by dennismunene** [comment]
@parasharrajat Tested sending links , It works fine.

**Comment by dennismunene** [comment]
The link is shown , in this case ``https://google.com``

**Comment by mountiny** [comment]
Feel free to submit the PR @dennismunene, thank you for the proposal and thanks @parasharrajat for the review!

I am surprised that the normal links work fine, hopefully, it will work well once tested.

**Comment by dennismunene** [comment]
@mountiny @parasharrajat 
I double checked sending links , This does replace links. I've updated my proposal with the appropriate fix. 
Only ``a`` tags with this prop ``data-expensify-source`` should be considered attachments.

**Comment by dennismunene** [comment]
Applied for the Job on Upwork & Raised [PR](https://github.com/Expensify/App/pull/8438) Ready for Review

**Comment by marcaaron** [comment]
>Newly uploaded MP4 file should show mp4 tag

How does the proposal address this? It looks like it will ensure this does not happen ever vs. inconsistently like shown in the video.

**Comment by dennismunene** [comment]
@marcaaron All attachments are currently represented with the ``[Attachment]`` Tag , Should I introduce a new one for Mp4 ? e.g ``[[mp4]]`` . If so ,whats the correct pattern to represent other file types like pdf,html... etc ?

**Comment by parasharrajat** [comment]
> Newly uploaded MP4 file should show mp4 tag

My mistake. I didn't ask for clarification. I went with what the title is saying and as the attachment tag is shown on the video upload, I thought it should be the case always.

**Comment by mountiny** [comment]
Oh, I missed clarifying the Expected section too. I assumed we want to always show the `[Attachement]` for any type of attachment which the proposal does, right? 


I will update the body of the issue.

**Comment by dennismunene** [comment]
@mountiny Yes the proposal covers this , all attachments will be represented as `[Attachment]`

**Comment by marcaaron** [comment]
Sorry, was busy with some other things. Since this was reported by Applause I'd probably ask them if they are using a regression test to assume this behavior? If so, we can update whatever test they are using so this won't get reported again once fixed.

**Comment by mountiny** [comment]
@marcaaron Good call!

@kbecciv Would you be able to have a look at the regression tests that led to creating this issue and make sure they are updated with the correct `expected behaviour`? We want to make sure the `[Attachement]` is shown consistently for whatever extension.

**Comment by kbecciv** [comment]
@mountiny Sure, let me pass all information to QA team and we will retest it.

**Comment by mountiny** [comment]
@kbecciv Thank you very much! There was actually an inconsistency here which we are fixing, but for future this would be great to get straight!

**Comment by mountiny** [comment]
PR is almost merged 🎉

**Comment by dennismunene** [comment]
Still haven't received Offer on Upwork for this. This is my profile  https://www.upwork.com/freelancers/~017508f1bfdd7ba8d9

**Comment by mountiny** [comment]
cc @kevinksullivan

**Comment by kevinksullivan** [comment]
Hi @dennismunene , apologies for the delay, I was on vacation. Just sent you an offer. @parasharrajat invited you to the job for C+ as well.

**Comment by mvtglobally** [comment]
Issue not reproducible during KI retests. (Second week)

**Comment by dennismunene** [comment]
waiting for payment on this cc @kevinksullivan

**Comment by kevinksullivan** [comment]
Paid @dennismunene , thanks. @parasharrajat can you accept the offer for the job and I'll pay out for c+?

**Comment by mvtglobally** [comment]
Issue not reproducible during KI retests. (Third week)

**Comment by kevinksullivan** [comment]
Alright, offer sent @parasharrajat once you accept I'll take care of this and close out.
